src/propagators/text_map_codec.js

```diff
@@ -56,21 +57,28 @@ export default class TextMapCodec {
     }
 
     extract(carrier: any): ?SpanContext {
-        let header = carrier[this._contextKey];
-        let spanContext: ?SpanContext = null;
-        if (header) {
-            spanContext = SpanContext.fromString(this._decodedValue(header));
-            let baggage = {};
-            for (let key in carrier) {
-                if (carrier.hasOwnProperty(key) && Utils.startsWith(key, this._baggagePrefix)) {
+        // $FlowIgnore - I just want an empty span context.
+        let spanContext = new SpanContext();
+        let baggage = {};
+
+        for (let key in carrier) {
+            if (carrier.hasOwnProperty(key)) {
+                if (key === this._contextKey) {
+                    spanContext = SpanContext.fromString(this._decodedValue(carrier[this._contextKey]));
+                }
+
+                if (Utils.startsWith(key, this._baggagePrefix)) {
                     let keyWithoutPrefix = key.substring(this._baggagePrefix.length);
-                    let value = this._decodedValue(carrier[key]);
-                    baggage[keyWithoutPrefix] = value;
+                    baggage[keyWithoutPrefix] = this._decodedValue(carrier[key]);
                 }
             }
-            spanContext.baggage = baggage;
         }
 
+        if (constants.JAEGER_DEBUG_HEADER in carrier) {
```

Comment:
similarly, it is safer to have this inside the loop and compare against lowerCase(key)